lib/iex/lib/iex/helpers.ex

```diff
@@ -440,6 +443,17 @@ defmodule IEx.Helpers do
     dont_display_result()
   end
 
+  # Given any "term", this function returns all the protocols in
+  # :code.get_path() implemented by the data structure of such term, in the form
+  # of a binary like "Protocol1, Protocol2, Protocol3".
+  defp all_implemented_protocols_for_term(term) do
+    :code.get_path()
+    |> Protocol.extract_protocols()
+    |> Enum.uniq()
+    |> Enum.reject(fn(protocol) -> is_nil(protocol.impl_for(term)) end)
```

Comment:
@josevalim I explicitly used `is_nil/1` so that it's very clear what's going on instead of relying on truthyness/falseyness. :)